golem-worker-service-base/src/service/template/error.rs

```diff
@@ -0,0 +1,120 @@
+use golem_api_grpc::proto::golem::worker::{
+    self, worker_error, worker_execution_error, UnknownError, WorkerError as GrpcWorkerError,
+};
+use golem_service_base::service::auth::AuthError;
+use tonic::Status;
+
+// The dependents of golem-worker-service-base is expected
+// to have a template service internally that can depend on this base error
+#[derive(Debug, thiserror::Error)]
+pub enum TemplateServiceError {
+    #[error(transparent)]
+    Auth(#[from] AuthError),
+    #[error("Bad Request: {0:?}")]
+    BadRequest(Vec<String>),
+    #[error("Unauthorized: {0}")]
+    Unauthorized(String),
+    #[error("Limit Exceeded: {0}")]
+    LimitExceeded(String),
```

Comment:
Are we bumping into this error in OSS?

Reply:
i think it should not be in OSS, but not sure, if with this approach it possible to do it better, 
maybe here should be some Custom error, which will be specially handled in cloud version

Reply:
These errors are the same as the grpc errors

https://github.com/golemcloud/golem/blob/main/golem-api-grpc/proto/golem/template/template_error.proto